### ISSUE TITLE
how to integrate pyimgui

### ISSUE THREAD

**Issue by Choons** [panxinmiao/three-py#37]
Hi, I really like three-py, but I miss having the simple UI functionality of three.js. Dear Imgui is the most similar to what three.js uses and python bindings exist for it as pyimgui, but I'm not sure how to go about using it in three-py. If it's not too difficult, I would appreciate some guidance on how to use it.

**Comment by panxinmiao** [comment]
I’ve integrated Imgui into wgpu-py (https://github.com/pygfx/wgpu-py/pull/516). which can be used in three-py via `wgpu.utils.imgui.ImguiRenderer`. You can refer to the [example](https://github.com/pygfx/wgpu-py/blob/main/examples/imgui_renderer_sea.py) there to see how to use it directly in three-py.

Additionally, I’ve shifted some of my work to the [Pygfx](https://github.com/pygfx/pygfx) project (another similar Python 3D rendering engine), where I’ve implemented features like skinning and morph target animations, keyframe interpolation, some advanced rendering, a glTF loader, and better support for glTF models. Currently, pygfx might offer better maintenance and support, so I recommend giving it a try.  (Uh, Perhaps one day three-py will receive further updates and new capabilities—let’s hope so, haha!).

**Edit:** 
I just realized that the version of wgpu-py currently used by three-py is below 0.16.0, while ImguiRenderer is available only in wgpu-py 0.17.3 and above. I’ll find some time to update three-py to the latest version of wgpu and add an example using ImguiRenderer. However, I’ve been a bit busy lately, so it might take some time :)

**Comment by Choons** [comment]
Oh great! I actually found pygfx before three-py, but I liked how three-py worked more at the time because I am familiar with three.js and your examples were better. Thanks for the reply, and I look forward to using imgui if it's not too much trouble for you to update the wgpu-py back end. I'll also check out the progress pygfx has made in the couple of years since I last used it.

Edit:
I tested three-py running with wgpu 0.18.1 and all the examples worked except for morph.py. When running the examples I got these warnings: 

No config found!
EGL says it can present to the window but not natively
Max vertex attribute stride unknown. Assuming it is 2048

and when I closed the examples by clicking the corner x, the windows closed normally but then the console reported a segmentation fault. It didn't seem to affect any subsequent runs. My system is Ubuntu running X11 instead of Wayland because it works better with my Nvidia GPU. At any rate, it looks like three-py can use the latest wgpu with minimal problems. 

I tried to figure out how to use wgpu.utils.imgui.ImguiRenderer as you suggested, but it's not clear to me how to modify a three-py example to use it. I hope you find this info useful.

I also looked at pygfx again and remembered the reason I like three-py better is because of the easy access to the render loop. It seems not so easy to do that & control timing in pygfx.

Edit 2: I installed the latest pygfx version and noticed it does the same config/EGL warnings and seg fault on closing windows so that must be an issue with wgpu-py and/or the python GLFW module instead of something specific to three-py + new wgpu-py
